src/EnergyPlus/BoilerSteam.cc

```diff
@@ -196,15 +197,13 @@ namespace BoilerSteam {
             thisBoiler.Name = state.dataIPShortCut->cAlphaArgs(1);
 
             // Validate fuel type input
-            bool FuelTypeError(false);
-            UtilityRoutines::ValidateFuelTypeWithAssignResourceTypeNum(
-                state.dataIPShortCut->cAlphaArgs(2), thisBoiler.BoilerFuelTypeForOutputVariable, thisBoiler.FuelType, FuelTypeError);
-            if (FuelTypeError) {
+            thisBoiler.FuelType = static_cast<Constant::eResource>(
```

Comment:
Not sure if we want to do this in this PR or not, but would be good to convert all of these FuelType variables from `eResource` to `eFuel`.

Reply:
> Not sure if we want to do this in this PR or not, but would be good to convert all of these FuelType variables from `eResource` to `eFuel`.

@dareumnam @amirroth  I must be missing something. It appears that the subset enums like `eFuel` are not being used anywhere. They are declared and mapped to each other in `DataConstants` but I don't see them used elsewhere. 

In this case here, `ValidateFuelTypeWithAssignResourceTypeNum` rejected any input that was not a fuel type, but now, using `getEnumerationValue(Constant::eResourceNamesUC . . .` will accept any resource type. The key choices in the IDD will reject any non-fuels, but do we want to rely on that being the only gatekeeper?

Reply:
@mjwitte Thanks for your insight. As you said, `eFuel` has not been used yet in this PR, but I also think we should convert all FuelType variables from `eResource` to `eFuel`, not just rely on InputProcessor to reject non-fuels. After this PR, I'll open a new PR for that.

Reply:
Or I can do that in this PR either.

Reply:
> @mjwitte Thanks for your insight. As you said, `eFuel` has not been used yet in this PR, but I also think we should convert all FuelType variables from `eResource` to `eFuel`, not just rely on InputProcessor to reject non-fuels. After this PR, I'll open a new PR for that.

> Or I can do that in this PR either.

@dareumnam Given the amount of changes here, I would finish this PR and start a new one. Be sure to explain this in the top comment on the PR.